### PR TITLE
[internal/components] register jaegerremotesampling extension

### PR DESCRIPTION
**Pull request by frzifus**
**Description:** <Describe what has changed.>

Register jaegerremotesampling extension

**Link to tracking Issue:** <Issue number if applicable>
- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/5749

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: internal/components/components.go [hunk 1]
```diff
@@ -208,6 +209,7 @@ func Components() (otelcol.Factories, error) {
 		oidcauthextension.NewFactory(),
 		sigv4authextension.NewFactory(),
 		zpagesextension.NewFactory(),
+		jaegerremotesampling.NewFactory(),
```
Comment (crobert-1) [inline internal/components/components.go:212]: Nit: These are generally in alphabetical order, so it'd be nice to move this up to be in its proper place. I see a couple out of place already, but it'd still be nice.

**Comment by atoulme** [comment]
Please run `make crosslink`

**Comment by frzifus** [comment]
done :+1:

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
> go.mod/go.sum deps changes detected, please run "make gotidy" and commit the changes in this PR.

**Comment by jpkrohling** [comment]
Looks like it's still failing:

```
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
diff --git a/go.mod b/go.mod
index c01ef5f..c9940cb 100644
--- a/go.mod
+++ b/go.mod
@@ -168,7 +168,6 @@ require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/windowsperfcountersreceiver v0.81.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zipkinreceiver v0.81.0
 	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/zookeeperreceiver v0.81.0
-	github.com/prometheus/client_golang v1.16.0
 	go.opentelemetry.io/collector v0.81.0
 	go.opentelemetry.io/collector/exporter v0.81.0
 	go.opentelemetry.io/collector/exporter/loggingexporter v0.81.0
@@ -542,6 +541,7 @@ require (
 	github.com/pmezard/go-difflib v1.0.1-0.20181226105442-5d4384ee4fb2 // indirect
 	github.com/power-devops/perfstat v0.0.0-20220216144756-c35f1ee13d7c // indirect
 	github.com/pquerna/cachecontrol v0.1.0 // indirect
+	github.com/prometheus/client_golang v1.16.0 // indirect
 	github.com/prometheus/client_model v0.4.0 // indirect
 	github.com/prometheus/common v0.44.0 // indirect
 	github.com/prometheus/common/sigv4 v0.1.0 // indirect
go.mod/go.sum deps changes detected, please run "make gotidy" and commit the changes in this PR.

```

**Pull request merged**